File: src/main/java/optimizer/LinearEquation.java
// LinearEquation.java
//
// Copyright 2002-2022 Jack Boyce and the Juggling Lab contributors

package jugglinglab.optimizer;


public class LinearEquation {
    protected int numVars;  // number of variables
    protected double[] coefs;  // coefficients in linear equation
    protected boolean done;  // flag for use with optimizer


    public LinearEquation(int vars) {
        numVars = vars;
        coefs = new double[vars + 1];
        done = false;
    }

    public void setCoefficients(double[] c) {
        coefs = c;
    }

    public double coef(int col) {
        return coefs[col];
    }

    public double constant() {
        return coefs[numVars];
    }

    public boolean done() {
        return done;
    }

    public void setDone(boolean d) {
        done = d;
    }
}

File: composeApp/src/jvmMain/kotlin/jugglinglab/optimizer/LinearEquation.kt
//
// LinearEquation.kt
//
// Copyright 2002-2026 Jack Boyce and the Juggling Lab contributors
//

package jugglinglab.optimizer

class LinearEquation {
    var numVars: Int  // number of independent variables
    var coefs: DoubleArray  // coefficients in linear equation
    var done: Boolean  // flag for use with optimizer

    constructor(vars: Int) {
        numVars = vars
        coefs = DoubleArray(vars + 1)
        done = false
    }

    fun setCoefficients(c: DoubleArray) {
        coefs = c
    }

    fun coef(col: Int): Double {
        return coefs[col]
    }

    fun constant(): Double {
        return coefs[numVars]
    }
}
